msal/application.py

```diff
@@ -69,6 +70,36 @@ def _clean_up(result):
     return result
 
 
+def _preferred_browser():
+    """Register Edge and return a name suitable for webbrowser.get(...),
+    if running on Linux and there is no BROWSER env var, otherwise return None.
+    """
+    browser_path = "/usr/bin/microsoft-edge"  # Use a full path owned by sys admin
+    user_has_no_preference = "BROWSER" not in os.environ
+    user_opted_in_for_edge = "microsoft-edge" in os.environ.get("BROWSER", "")
+    if ((user_opted_in_for_edge or user_has_no_preference)
+            and sys.platform == "linux"  # On Linux, only Edge will have CA support
+            and os.path.exists(browser_path)):  # Edge is usually installed here
+        try:
+            import webbrowser  # Lazy import. Some distro may not have this.
+            browser_name = "msal-edge"  # A unique name rather than "microsoft-edge"
+                # otherwise `BROWSER="microsoft-edge"; webbrowser.get("microsoft-edge")`
+                # would return a GenericBrowser instance which won't work.
+            try:
+                registration_available = isinstance(
+                    webbrowser.get(browser_name), webbrowser.BackgroundBrowser)
+            except webbrowser.Error:
+                registration_available = False
+            if not registration_available:
+                logger.debug("Register %s with %s", browser_name, browser_path)
```

Comment:
You don't need `registration_available`, because the exception tells you whether the browser is registered. And do you need to check the registered type? Seems reasonable to assume applications won't register their own "msal-edge".

```suggestion
            try:
                webbrowser.get(browser_name)
                return browser_name
            except webbrowser.Error:
                logger.debug("Register %s with %s", browser_name, browser_path)
```

Reply:
Due to the significance of the need for `BackgroundBrowser`, as explained in the topic above, such an `isinstance(...)` is used here. Perhaps we could rename `registration_available` to something like `suitable_registration_available`, to emphasize our intention here.